Sources/NIOKit/Future+Optional.swift

```diff
@@ -0,0 +1,75 @@
+import NIO
+
+extension EventLoopFuture {
+    
+    /// Calls a closure on an optional value in an `EventLoopFuture` if it exists.
+    ///
+    ///     let optional = eventLoop.future(Optional<Int>.some(42))
+    ///     let some = optional.flatMapOptionalThrowing { int -> Float in
+    ///         return int * 3.14
+    ///     }
+    ///     // some: EventLoopFuture(Optional(131.88))
+    ///
+    /// - parameters:
+    ///   - closure: The closure to call on the unwrapped optional value.
+    ///   - unwrapped: The optional's value, unwrapped.
+    /// - returns: The result of the closure if the optional was unwrapped, or nil if it wasn't, wrapped in an `EventLoopFuture`.
+    public func flatMapOptionalThrowing<Wrapped, Result>(_ closure: @escaping (_ unwrapped: Wrapped)throws -> Result?) -> EventLoopFuture<Result?>
+        where Value == Optional<Wrapped>
+    {
+        return self.flatMapThrowing { optional in
+            return try optional.flatMap(closure)
+        }
+    }
+    
+    
+    /// Calls a closure on an optional value in an `EventLoopFuture` if it exists.
+    ///
+    ///     let optional = eventLoop.future(Optiona<Int>.some(42))
+    ///     let some = optional.mapOptional { int -> EventLoopFuture<Float> in
+    ///         return int * 3.14
+    ///     }
+    ///     // some: EventLoopFuture(Optional(131.88))
+    ///
+    /// - parameters:
+    ///   - closure: The closure to call on the unwrapped optional value.
+    ///   - unwrapped: The optional's value, unwrapped.
+    /// - returns: The result of the closure if the optional was unwrapped, or nil if it wasn't, wrapped in an `EventLoopFuture`.
+    public func flatMapOptional<Wrapped, Result>(
+        _ closure: @escaping (_ unwrapped: Wrapped)throws -> EventLoopFuture<Result>
```

Comment:
Same whitespace thing here
```suggestion
        _ closure: @escaping (_ unwrapped: Wrapped) throws -> EventLoopFuture<Result>
```